src/main/java/com/core/Response.java

```diff
@@ -0,0 +1,207 @@
+        Printer.printError("The description of a deadline cannot be empty");
+        return true;
+    }),
+    DEADLINE_NO_TIME("(?i)^deadline (((?!/by).)+$)|(.+ /by\\s*$)", (i, s) -> {
+        Printer.printError("The due date of a deadline cannot be empty");
+        return true;
+    }),
+    DEADLINE("(?i)^deadline .+ /by .+", (i, s) -> {
+        String[] parts = splitTwoDelimiters(i, "(?i)^deadline ", "(?i)/by ");
+        addTask(new Deadline(parts[0], DateTime.parseString(parts[1])), s);
+        return true;
+    }),
+    DEADLINE_WRONG_TIME("(?i)^deadline .+ /by .+", (i, s) -> {
+        Printer.printError("The date must be in the format 'DD/MM/YYYY HHMM'");
+        return true;
+    }),
+    UNKNOWN(".*", (i, s) -> {
+        Printer.printError("I'm sorry but I don't know what that means :-(");
+        return true;
+    });
+
+    private String regex;
+    private ResponseFunc func;
+
+    Response(String r, ResponseFunc f) {
+        regex = r;
+        func = f;
+    }
+
+    /**
+     * Given a string and program state, if string matches regex this enum will call its response
+     * function.
+     *
+     * @param i input string
+     * @param s state object
+     * @return boolean if the string has matched
+     */
+    public boolean call(String i, State s) {
+        if (i.matches(regex)) {
+            return func.funcCall(i, s);
+        }
+        return false;
+    }
+
+    /**
+     * Assuming input is ".+ [0-9]+", it splits at whitespace and returns an Integer of the second
+     * part of the string.
+     *
+     * @param input input string
+     * @return integer of second part of string
+     */
+    private static int getNumber(String input) {
+        return Integer.parseInt(input.split(" ", 2)[1]);
+    }
+
+    /**
+     * Given an index and state object, check if index in bounds of list in state object.
+     *
+     * @param index index to check
+     * @param s     state object
+     * @return False if out of bounds
+     */
+    private static boolean checkValidIndex(int index, State s) {
+        if (index < 0 || index > s.list.size() - 1) {
+            Printer.printError("That is not a valid task index");
+            return false;
+        }
+        return true;
+    }
+
+    /**
+     * Assuming input is "(head).*(mid).*" returns the two texts between them.
+     *
+     * @param input input string
+     * @param head  head regex match
+     * @param mid   mid regex match
+     * @return array of two text parts
+     */
+    private static String[] splitTwoDelimiters(String input, String head, String mid) {
+        String[] parts = input.split(mid, 2);
+        parts[0] = parts[0].split(head)[1];
+        return parts;
+    }
+
+    /**
+     * Insert task into list and prints message string.
+     *
+     * @param t task to be added
+     * @param s state object
+     */
+    private static void addTask(DoableTask t, State s) {
+        s.list.add(t);
```

Comment:
Another way would be to abstract this step (same for deleting, marking a task done etc) out and let the State object handle modifications to the list of tasks; this would help Response focus on the console outputs.